src/Component/Filter/AttributeCombo/AttributeCombo.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+
+import { Select } from 'antd';
+import FormItem from 'antd/lib/form/FormItem';
+const Option = Select.Option;
+
+/**
+ * Combobox offering the attributes to be filtered on.
+ */
+class AttributeCombo extends React.Component<any, any> {
+
+  // PROPS
```

Comment:
React usually defines its props with [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html).
As we make use of TypeScript [another approach with interfaces](https://charleslbryant.gitbooks.io/hello-react-and-typescript/content/TypeScriptAndReact.html) should be the way to go.

The props should not be declared as member variables. The declared member variables will not be associated with the props. So you will have `this.internalDataDef` and `this.props.internalDataDef` and so on.

Reply:
Good point, I have this on my list. This is the next thing I will tackle.